pkg/extensions/tbtc/tbtc.go

```diff
@@ -128,6 +166,8 @@ func (t *tbtc) monitorProvideRedemptionSignature(
 	actBackoffFn backoffFn,
 	timeout time.Duration,
 ) error {
+	initialDepositState := chain.AwaitingWithdrawalSignature
+
 	monitoringStartFn := func(
```

Comment:
Not for this PR and probably not for this release but I wonder whether we shouldn't wait e.g. 6 blocks before we start the monitoring. If we do, we can avoid dangling goroutines. On the other hand, even if the transaction was dropped, it should soon be included in another block as the application should keep tracking deposit progress. Just something to note and think about.